src/components/Agreements/ViewAgreement/Sections/Finances.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import { Accordion, Badge, Button, Icon } from '@folio/stripes/components';
+
+import FinancesAgreementLines from './FinancesAgreementLines';
+
+export default class Finances extends React.Component {
+  static propTypes = {
+    financesAgreementLines: PropTypes.arrayOf(PropTypes.object),
+    fetchMoreEresources: PropTypes.func,
+    id: PropTypes.string,
+    onToggle: PropTypes.func,
+    open: PropTypes.bool,
+  };
+
+  renderBadge = () => {
+    const count = get(this.props.financesAgreementLines, ['length']);
+    if (count === undefined) {
+      return <Icon icon="spinner-ellipsis" width="10px" />;
+    }
+    return <Badge data-test-finances-agreement-lines-count={count}>{count}</Badge>;
+  }
+
+  render() {
+    const { id, onToggle, open } = this.props;
+
+    const buttonAndBadge =
+      <div>
+        <Button align="end">
+          <Icon icon="edit">
+            <FormattedMessage id="ui-agreements.agreements.editFinancesAgreementLines" />
+          </Icon>
+        </Button>
+        {' '}
+        {this.renderBadge()}
+      </div>;
+
+    return (
+      <Accordion
+        id={id}
+        label={<FormattedMessage id="ui-agreements.agreements.financesAgreementLines" />}
+        open={open}
+        onToggle={onToggle}
+        displayWhenClosed={buttonAndBadge}
```

Comment:
Are you sure we're rendering both the button and badge in both scenarios (open/closed)? I was under the impression that we render button when open and badge when closed. Gill can probably confirm one way or another.

Reply:
In this pull request (UIER-32) I did not change the positioning and visibility of the edit button. That was done in https://issues.folio.org/browse/UIER-31 . I only moved the button code from FinancesAgreementLines.js to Finances.js.

Reply:
If it is an issue it can be reported in UIER-31 or in a separate jira ticket.

Reply:
Files as https://issues.folio.org/browse/UIER-54